src/Native/System.Native/pal_networking.cpp

```diff
@@ -2394,7 +2421,11 @@ static Error WaitForSocketEventsInner(int32_t port, SocketEvent* buffer, int32_t
     {
         // This copy is made deliberately to avoid overwriting data.
         struct kevent evt = events[i];
+#if KEVENT_REQUIRES_INT_PARAMS
         buffer[i] = {.Data = reinterpret_cast<uintptr_t>(evt.udata), .Events = GetSocketEvents(evt.filter, evt.flags)};
+#else
+        buffer[i] = {.Data = static_cast<unsigned long>(evt.udata), .Events = GetSocketEvents(static_cast<short>(evt.filter), static_cast<unsigned short>(evt.flags))};
+#endif
```

Comment:
Can a similar thing be done with KEVENT_REQUIRES_INT_PARAMS as was done with KEVENT_HAS_VOID_UDATA to avoid duplicating each of these three sections of code simply for a cast?